starters/next-wordpress-starter/pages/api/preview.js

```diff
@@ -0,0 +1,45 @@
+import { getPostPreview } from '../../lib/Posts';
+import { getPagePreview } from '../../lib/Pages';
+import { getAuthCredentials } from '../../lib/WordPressClient';
+
+const preview = async (req, res) => {
+	const { secret, uri, id, content_type } = req.query;
+
+	if (!secret || !uri || !id || !content_type) {
+		return res.redirect('/500');
+	}
+
+	if (secret !== process.env.PREVIEW_SECRET || !uri) {
+		return res.redirect('/500');
+	}
+
+	if (content_type !== 'posts' && content_type !== 'pages') {
+		return res.redirect('/500');
+	}
+
+	const credentials = getAuthCredentials();
+	if (content_type === 'posts') {
+		const post = await getPostPreview(id, credentials);
+		if (!post) {
+			return res.redirect('/500');
+		}
+
+		res.setPreviewData({
+			key: id,
+		});
+	}
+	if (content_type === 'pages') {
+		const page = await getPagePreview(id, credentials);
+		if (!page) {
+			return res.redirect('/500');
+		}
+
+		res.setPreviewData({
+			key: id,
+		});
+	}
```

Comment:
I think this is fine for now but it doesn't feel very scaleable, although maybe it doesn't need to be. I'm wondering if there is a way to make the `getXPreview` more dynamic based on the content type, but if we are only supporting page and post (as I'm pretty sure that is all that WP supports out of the box), it's fine.

Reply:
Yeah, I thought about that too, but it felt like a premature abstraction, because of the 2+ rule to extract, but let me know if you prefer to do that!

Reply:
OK that is fair! Let's keep it like this for now. We probably won't need to add more content types until we add support for the WP Advanced Custom Fields plugin.